test/types.js

```diff
@@ -188,24 +189,28 @@ exports.numrange = {
   ]
 }
 
+function toPostgresInterval (obj) {
+  const base = Object.create(PostgresInterval.prototype)
+  return Object.assign(base, obj)
+}
 exports.interval = {
   format: 'text',
   id: 1186,
   tests: [
     ['01:02:03', function (t, value) {
       t.equal(value.toPostgres(), '3 seconds 2 minutes 1 hours')
-      t.deepEqual(value, { hours: 1, minutes: 2, seconds: 3 })
+      t.deepEqual(value, toPostgresInterval({ years: 0, months: 0, days: 0, hours: 1, minutes: 2, seconds: 3, milliseconds: 0 }))
```

Comment:
This is probably the only meaningful change, as deepEqual doesn't ignore objects' prototypes. So I added a small function to fix that.